postgres/toddl.go

```diff
@@ -225,3 +228,32 @@ func cvtForeignKeys(conv *internal.Conv, srcTable string, srcKeys []schema.Forei
 	}
 	return spKeys
 }
+
+func cvtIndexes(conv *internal.Conv, spTableName string, srcTable string, srcIndexes []schema.Index, schemaIndexKeys map[string]bool) []ddl.CreateIndex {
+	var spIndexes []ddl.CreateIndex
+	for _, srcIndex := range srcIndexes {
+		var spKeys []ddl.IndexKey
+		for _, k := range srcIndex.Keys {
+			spCol, err := internal.GetSpannerCol(conv, srcTable, k.Column, true)
+			if err != nil {
+				conv.Unexpected(fmt.Sprintf("Can't map index key column name for table %s", srcTable))
+				continue
+			}
+			spKeys = append(spKeys, ddl.IndexKey{Col: spCol, Desc: k.Desc})
+		}
+		if srcIndex.Name == "" {
+			// Generate a name if index name is empty in Postgres.
+			// Collision of index name will be handled by ToSpannerIndexKey.
+			srcIndex.Name = fmt.Sprintf("Index_%s", srcTable)
+		}
+		spKeyName := internal.ToSpannerIndexKey(srcIndex.Name, schemaIndexKeys)
```

Comment:
As noted in mapping.go, how about we make the naming a bit more precise.
Suggest ToSpannerIndexName, and changing the var name to spIndexName.
And maybe schemaIndexKeys to spIndexNames?

Or is there some reason to keep using "Keys" here?
I find it confusing/misleading, since we have other kinds of keys here (the columns themselves).


be consistent about terminology e.g. index key vs index key name vs constraint name; maybe we should use index name everywhere, including the function name, the comment and in README.md?

Reply:
updated function name as mentioned and parameter name as well